src/hostapi/pulseaudio/pa_linux_pulseaudio.c

```diff
@@ -931,7 +932,7 @@ PaError PaPulseAudio_BlockingInitRingBuffer( PaUtilRingBuffer * rbuf,
 
     if( ringbufferBuffer == NULL )
     {
-        PA_PULSEAUDIO_SET_LAST_HOST_ERROR( 0,
+        PA_PULSEAUDIO_SET_LAST_HOST_ERROR( paInsufficientMemory,
```

Comment:
The only time you should be calling `PA_PULSEAUDIO_SET_LAST_HOST_ERROR` is if you return `paUnanticipatedHostError`. If you can map to a generic PortAudio error, as in this case `paInsufficientMemory`, then you should not call `PA_PULSEAUDIO_SET_LAST_HOST_ERROR`.

Reply:
I've corrected errors as requested

Reply:
There is still a problem.

        PA_PULSEAUDIO_SET_LAST_HOST_ERROR( 0,
                                           "PaPulseAudio_BlockingInitRingBuffer: Not enough memory to handle request" );
        return paInsufficientMemory;

The code should not use PA_PULSEAUDIO_SET_LAST_HOST_ERROR when there is not a host error code.
If you are passing back a "pa" error then you should not set it.
You could use PA_DEBUG() instead of PA_PULSEAUDIO_SET_LAST_HOST_ERROR() if you want to log at that point.

Reply:
Also, always prefer returning PA error codes to `paUnanticipatedHostError`

Reply:
Ok now everything is PA_DEBUG and no more error reporting with `PA_PULSEAUDIO_SET_LAST_HOST_ERROR` . I have learn how `ENSURE_` works on OSS and ALSA to understand how they should behave but now it should be ok as no more `paUnanticipatedHostError`.